git/packet_writer.go

```diff
@@ -0,0 +1,124 @@
+package git
+
+import (
+	"io"
+
+	"github.com/github/git-lfs/tools"
+)
+
+type PacketWriter struct {
+	// buf is the internal buffer of bytes used to store data given in calls
+	// to Write that don't warrant a protocol write.
```

Comment:
Definitely confusing! Cleaned it up in https://github.com/github/git-lfs/pull/1622/commits/7be86e55842f9ca4805484a0163399845e1bbae8.